examples/export-data/main.ts

```diff
@@ -1,2 +1,183 @@
-// Run this with yarn export-data or npx tsx main.ts
-console.log("Hello World")
\ No newline at end of file
+import mongoose from 'mongoose';
+import dotenv from 'dotenv';
+import { dbConnect } from './utils/db-connect';
+import { AnalyticsViewer, AnalyticsLogger, AnalyticsManager } from "bog-analytics";
+import { ClickEventModel } from './models/click-event';
+import { VisitEventModel } from './models/visit-event';
+import { InputEventModel } from './models/input-event';
+import { CustomEventModel } from './models/custom-event';
+import { Project } from './utils/types';
+import { EventEnvironment } from './utils/types';
+import { getProjectIDByName, createProject } from './actions/project'
+
+dotenv.config();
+const CLIENT_KEY = process.env.CLIENT_API_SECRET || '';
+const SERVER_KEY = process.env.SERVER_API_SECRET || '';
+const projectName = "Bits of Good Example Project";
+
+const main = async () => {
+    try {
+        await dbConnect();
+        console.log('Connected to MongoDB');
+
+        // Create the project if it doesn't exist
+        let projectId = await getProjectIDByName(projectName);
+        if (!projectId) {
+            console.log(`Project '${projectName}' does not exist. Creating new project`);
+            
+            const newProject: Partial<Project> = {
+                projectName: projectName,
+                clientApiKey: CLIENT_KEY,
+                serverApiKey: SERVER_KEY,
+                privateData: false,
+                deletionPolicy: -1,
+            };
+            
+            const createdProject = await createProject(newProject);
+            projectId = createdProject._id;
+            console.log(`Created new project with ID: ${projectId}`);
+        } else {
+            console.log(`Project ID found: ${projectId}`);
+        }
+
+        const { clickEvents, visitEvents, inputEvents, customEvents } = await fetchEvents();
+        
+        console.log(`Fetched ${clickEvents?.length} click events`)
+        console.log(`Fetched ${visitEvents?.length} visit events`)
+        console.log(`Fetched ${inputEvents?.length} input events`)
+        console.log(`Fetched ${customEvents?.length} custom events`)
+
+        for (const event of clickEvents || []) {
+            await ClickEventModel.create({
```

Comment:
Edge case you'll want to take into account when fetching events: this script is going be run once a day so on day 1 bog analytics might have 100 total events, day 2 total 200 events, etc.

when its run on day 1, it copies over 100 events. when its run on day 2 it copies over 200 events. However, on day 2 it should only copy 100 events (not 200) because we already copied over the first 100 events on day 1. Basically, we want to prevent duplicates

What you need to do:
- query your database for the most recently created event
- get the createdAt date for that event
- Query bog analytics for events created after that date - bog analytics supports this functionality natively: https://github.com/GTBitsOfGood/bog-analytics/tree/main/package#getallclickeventsprojectname-string-aftertime-date-promiseclickevent--null
- If your database is empty then you don't need to input the afterDate field.